Tests/Benchmarks/RayFrustumBenchmark.cpp

```diff
@@ -0,0 +1,452 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <algorithm>
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include "Acts/Surfaces/BoundaryCheck.hpp"
+#include "Acts/Tests/CommonHelpers/BenchmarkTools.hpp"
+#include "Acts/Utilities/BoundingBox.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Frustum.hpp"
+#include "Acts/Utilities/Ray.hpp"
+#include "Acts/Utilities/Units.hpp"
+
+using namespace Acts;
+
+struct O {};
+using Box = Acts::AxisAlignedBoundingBox<O, float, 3>;
+using VertexType = Box::VertexType;
+using vertex_array_type = Box::vertex_array_type;
+using value_type = Box::value_type;
+using Frustum3 = Frustum<float, 3, 4>;
+
+bool optIntersect1(const Box& box, const Ray<float, 3>& ray) {
+  const VertexType& origin = ray.origin();
+  const vertex_array_type& idir = ray.idir();
+
+  vertex_array_type t0s = (box.max() - origin).array() * idir;
+  vertex_array_type t1s = (box.min() - origin).array() * idir;
+
+  vertex_array_type tsmaller = t0s.min(t1s);
+  vertex_array_type tbigger = t0s.max(t1s);
+
+  value_type tmin = tsmaller.maxCoeff();
+  value_type tmax = tbigger.minCoeff();
+
+  return tmin < tmax && tmax > 0.0;
+}
```

Comment:
This is only used for results comparison with naiveIntersect. Maybe a development artifact from an earlier version of this microbenchmark?

Reply:
Yeah I think this isn't used. True